# **Worldwide Travel Analysis**

## Notebook Objectives
With this notebook we plan to accomplish the following:

- Fetch dataset from Kaggle / Load dataset from local files

- Perform ETL: cleaning, transformation, and restructuring

- Engineer any usable features (e.g., temperature simplifications, duration categories)

- Explore dataset and produce preliminary analysis

- Build visualisations that support research questions

- Prepare cleaned data for dashboard and Streamlit app

## Inputs / Outputs

**Inputs:**
* The only input we need for our analysis is the raw ```worldwide_travel_cities.csv``` file.

**Outputs:**


The files, artefacts we plan to generated in this notebook include:

- A cleaned dataset of the ```worldwide_travel_cities.csv``` file. 

- Feature-engineered dataset

- Exported visualisation images

- Tables/statistics for dashboard or Streamlit app



---

## Section 1 — Import Dependencies

Section 1 content

In [83]:
import os
import pandas as pd
import numpy as np
import json
import matplotlib.pyplot as plt
import seaborn as sns
import kagglehub as kh
import ast

print('All libraries imported')

All libraries imported


---

## Section 2 — Exploring the Dataset & ETL Pipeline

### Section 2.1 - Loading Raw Dataset

In [84]:
# Load the original dataset
df_wta = pd.read_csv('data/raw_worldwide_travel_cities_dataset.csv')

print(f"Data loaded. Initial shape: {df_wta.shape}")

Data loaded. Initial shape: (560, 19)


### Section 2.2 - Cleaning the Dataset

Making a copy of the dataset to clean it.

In [85]:
# df is a copy of the original dataset where I'll do all my cleaning, processing etc 
df = df_wta.copy()
display(df.head())

id            city    country  \
0  c54acf38-3029-496b-8c7a-8343ad82785c           Milan      Italy   
1  0bd12654-ed64-424e-a044-7bc574bcf078  Yasawa Islands       Fiji   
2  73036cda-9134-46fc-a2c6-807782d59dfb        Whistler     Canada   
3  3872c9c0-6b6e-49e1-9743-f46bfe591b86      Guanajuato     Mexico   
4  e1ebc1b6-8798-422d-847a-22016faff3fd        Surabaya  Indonesia   

          region                                  short_description  \
0         europe  Chic streets lined with fashion boutiques, his...   
1        oceania  Crystal-clear waters, secluded beaches, and vi...   
2  north_america  Snow-capped peaks and lush forests create a se...   
3  north_america  Winding cobblestone streets and colorful facad...   
4           asia  Bustling streets filled with the aroma of loca...   

    latitude   longitude                                   avg_temp_monthly  \
0  45.464194    9.189635  {"1":{"avg":3.7,"max":7.8,"min":0.4},"2":{"avg...   
1 -17.290947  177.125786  {"1":{"avg":28,"max":30.8,"min":25.8},"2":{"av...   
2  50.117190 -122.954302  {"1":{"avg":-2.5,"max":0.4,"min":-5.5},"2":{"a...   
3  20.987700 -101.000000  {"1":{"avg":15.5,"max":22.8,"min":8.7},"2":{"a...   
4  -7.245972  112.737827  {"1":{"avg":28.1,"max":32.5,"min":25.5},"2":{"...   

                       ideal_durations budget_level  culture  adventure  \
0            ["Short trip","One week"]       Luxury        5          2   
1             ["Long trip","One week"]       Luxury        2          4   
2  ["Short trip","Weekend","One week"]       Luxury        3          5   
3  ["Weekend","One week","Short trip"]    Mid-range        5          3   
4             ["Short trip","Weekend"]       Budget        4          3   

   nature  beaches  nightlife  cuisine  wellness  urban  seclusion  
0       2        1          4        5         3      5          2  
1       5        5          2        3         4      1          5  
2       5        2          3        3         4      2          4  
3       3        1          3        4         3      4          2  
4       3        2          3        4         3      4          2

In [86]:
# using info() to see data types and missing values
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 560 entries, 0 to 559
Data columns (total 19 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 560 non-null    object 
 1   city               560 non-null    object 
 2   country            560 non-null    object 
 3   region             560 non-null    object 
 4   short_description  560 non-null    object 
 5   latitude           560 non-null    float64
 6   longitude          560 non-null    float64
 7   avg_temp_monthly   560 non-null    object 
 8   ideal_durations    560 non-null    object 
 9   budget_level       560 non-null    object 
 10  culture            560 non-null    int64  
 11  adventure          560 non-null    int64  
 12  nature             560 non-null    int64  
 13  beaches            560 non-null    int64  
 14  nightlife          560 non-null    int64  
 15  cuisine            560 non-null    int64  
 16  wellness           560 non

In [87]:
df.describe(include='all')

id     city        country  region  \
count                                    560      560            560     560   
unique                                   560      559            167       7   
top     c54acf38-3029-496b-8c7a-8343ad82785c  Granada  United States  europe   
freq                                       1        2             50     177   
mean                                     NaN      NaN            NaN     NaN   
std                                      NaN      NaN            NaN     NaN   
min                                      NaN      NaN            NaN     NaN   
25%                                      NaN      NaN            NaN     NaN   
50%                                      NaN      NaN            NaN     NaN   
75%                                      NaN      NaN            NaN     NaN   
max                                      NaN      NaN            NaN     NaN   

                                        short_description    latitude  \
count                                                 560  560.000000   
unique                                                560         NaN   
top     Chic streets lined with fashion boutiques, his...         NaN   
freq                                                    1         NaN   
mean                                                  NaN   22.502186   
std                                                   NaN   27.980022   
min                                                   NaN  -54.807306   
25%                                                   NaN    5.268054   
50%                                                   NaN   31.793618   
75%                                                   NaN   43.673199   
max                                                   NaN   78.719852   

         longitude                                   avg_temp_monthly  \
count   560.000000                                                560   
unique         NaN                                                545   
top            NaN  {"1":{"avg":28.5,"max":32.5,"min":25.2},"2":{"...   
freq           NaN                                                  3   
mean      7.914665                                                NaN   
std      78.813803                                                NaN   
min    -175.201808                                                NaN   
25%     -64.439118                                                NaN   
50%      10.711854                                                NaN   
75%      50.020162                                                NaN   
max     179.332896                                                NaN   

                            ideal_durations budget_level    culture  \
count                                   560          560  560.00000   
unique                                   21            3        NaN   
top     ["Short trip","Weekend","One week"]    Mid-range        NaN   
freq                                    120          339        NaN   
mean                                    NaN          NaN    3.85000   
std                                     NaN          NaN    0.81291   
min                                     NaN          NaN    2.00000   
25%                                     NaN          NaN    3.00000   
50%                                     NaN          NaN    4.00000   
75%                                     NaN          NaN    4.00000   
max                                     NaN          NaN    5.00000   

         adventure      nature     beaches   nightlife     cuisine  \
count   560.000000  560.000000  560.000000  560.000000  560.000000   
unique         NaN         NaN         NaN         NaN         NaN   
top            NaN         NaN         NaN         NaN         NaN   
freq           NaN         NaN         NaN         NaN         NaN   
mean      3.178571    3.728571    2.380357    3.019643    3.792857   
std       0.798190    0.903920    1.435547    0.921599

**The ID Column**

The ```id``` column contains 560 unique values, one for each row and it carries no meaning. Using the default dataframe index serves the same purpose and is much cleaner.

In [88]:
# dropping 'id' column as it's not needed for analysis
df.drop(columns=['id'], inplace=True)
display(df.head())

city    country         region  \
0           Milan      Italy         europe   
1  Yasawa Islands       Fiji        oceania   
2        Whistler     Canada  north_america   
3      Guanajuato     Mexico  north_america   
4        Surabaya  Indonesia           asia   

                                   short_description   latitude   longitude  \
0  Chic streets lined with fashion boutiques, his...  45.464194    9.189635   
1  Crystal-clear waters, secluded beaches, and vi... -17.290947  177.125786   
2  Snow-capped peaks and lush forests create a se...  50.117190 -122.954302   
3  Winding cobblestone streets and colorful facad...  20.987700 -101.000000   
4  Bustling streets filled with the aroma of loca...  -7.245972  112.737827   

                                    avg_temp_monthly  \
0  {"1":{"avg":3.7,"max":7.8,"min":0.4},"2":{"avg...   
1  {"1":{"avg":28,"max":30.8,"min":25.8},"2":{"av...   
2  {"1":{"avg":-2.5,"max":0.4,"min":-5.5},"2":{"a...   
3  {"1":{"avg":15.5,"max":22.8,"min":8.7},"2":{"a...   
4  {"1":{"avg":28.1,"max":32.5,"min":25.5},"2":{"...   

                       ideal_durations budget_level  culture  adventure  \
0            ["Short trip","One week"]       Luxury        5          2   
1             ["Long trip","One week"]       Luxury        2          4   
2  ["Short trip","Weekend","One week"]       Luxury        3          5   
3  ["Weekend","One week","Short trip"]    Mid-range        5          3   
4             ["Short trip","Weekend"]       Budget        4          3   

   nature  beaches  nightlife  cuisine  wellness  urban  seclusion  
0       2        1          4        5         3      5          2  
1       5        5          2        3         4      1          5  
2       5        2          3        3         4      2          4  
3       3        1          3        4         3      4          2  
4       3        2          3        4         3      4          2

In this case, our dataset did not have any missing values, so the cleaning process was fast. The only thing I did was delete the ```id``` column, as itoffered zero analytical value.

### Section 2.3 - Tranforming the Data

**City Rating Column**

To support clearer comparison between cities, I created a new column that calculates the average rating across all nine rating categories. Using an average instead of a raw sum prevents bias, some cities might score high overall simply because the individual ratings add up, even if they have very low values in certain areas. The averaged score provides a more balanced, and interpretable measure of each city's overall travel appeal.

In [89]:
ratings = df[['culture', 'adventure', 'nature', 'beaches', 'nightlife',
              'cuisine', 'wellness', 'urban', 'seclusion']]

city_rating = ratings.sum(axis=1) / len(ratings.columns)
df['city_rating'] = city_rating.round(1)
display(df.head())

city    country         region  \
0           Milan      Italy         europe   
1  Yasawa Islands       Fiji        oceania   
2        Whistler     Canada  north_america   
3      Guanajuato     Mexico  north_america   
4        Surabaya  Indonesia           asia   

                                   short_description   latitude   longitude  \
0  Chic streets lined with fashion boutiques, his...  45.464194    9.189635   
1  Crystal-clear waters, secluded beaches, and vi... -17.290947  177.125786   
2  Snow-capped peaks and lush forests create a se...  50.117190 -122.954302   
3  Winding cobblestone streets and colorful facad...  20.987700 -101.000000   
4  Bustling streets filled with the aroma of loca...  -7.245972  112.737827   

                                    avg_temp_monthly  \
0  {"1":{"avg":3.7,"max":7.8,"min":0.4},"2":{"avg...   
1  {"1":{"avg":28,"max":30.8,"min":25.8},"2":{"av...   
2  {"1":{"avg":-2.5,"max":0.4,"min":-5.5},"2":{"a...   
3  {"1":{"avg":15.5,"max":22.8,"min":8.7},"2":{"a...   
4  {"1":{"avg":28.1,"max":32.5,"min":25.5},"2":{"...   

                       ideal_durations budget_level  culture  adventure  \
0            ["Short trip","One week"]       Luxury        5          2   
1             ["Long trip","One week"]       Luxury        2          4   
2  ["Short trip","Weekend","One week"]       Luxury        3          5   
3  ["Weekend","One week","Short trip"]    Mid-range        5          3   
4             ["Short trip","Weekend"]       Budget        4          3   

   nature  beaches  nightlife  cuisine  wellness  urban  seclusion  \
0       2        1          4        5         3      5          2   
1       5        5          2        3         4      1          5   
2       5        2          3        3         4      2          4   
3       3        1          3        4         3      4          2   
4       3        2          3        4         3      4          2   

   city_rating  
0          3.2  
1          3.4  
2          3.4  
3          3.1  
4          3.1

There were 9 separate rating columns (culture, nightlife, cuisine, etc.). Individually, they’re useful, but creating a single metric summarising a city's appeal (by averaging), helps:
- create a simple overall quality score
- rank cities more clearly
- support our plan to build a city recommendation using Streamlit

**Annual Average Temperature Column**

We decide to create a new column containing the average annual temperature for each city to make the climate data easier to analyse and compare across cities. The original ```avg_temp_monthly column``` stored temperatures in a JSON-like structure, which made it difficult to use directly in visualisations or statistical analysis. 

To simplify this, I parsed the JSON for each row, extracted the monthly average temperatures, and calculated the mean value across all twelve months. This provides a single, clean annual temperature metric for each city, making it more practical for exploring patterns, testing our hypotheses, and building clear visual insights.

In [90]:
# Creating a function to calculate annual average temperature
def calculate_annual_avg_temp(json_string):
    # Converts the JSON-like string into a dictionary
    data = json.loads(json_string)
    
    # Extracts all monthly average temperatures
    monthly_avgs = [month["avg"] for month in data.values()]
    
    # Returns the average (mean) of the 12 months
    return (sum(monthly_avgs) / len(monthly_avgs))

In [91]:
# Calculating annual average temperature and adding it as a new column
df['annual_avg_temp_(C)'] = df['avg_temp_monthly'].apply(calculate_annual_avg_temp).round(0).astype(int)

display(df.head())

city    country         region  \
0           Milan      Italy         europe   
1  Yasawa Islands       Fiji        oceania   
2        Whistler     Canada  north_america   
3      Guanajuato     Mexico  north_america   
4        Surabaya  Indonesia           asia   

                                   short_description   latitude   longitude  \
0  Chic streets lined with fashion boutiques, his...  45.464194    9.189635   
1  Crystal-clear waters, secluded beaches, and vi... -17.290947  177.125786   
2  Snow-capped peaks and lush forests create a se...  50.117190 -122.954302   
3  Winding cobblestone streets and colorful facad...  20.987700 -101.000000   
4  Bustling streets filled with the aroma of loca...  -7.245972  112.737827   

                                    avg_temp_monthly  \
0  {"1":{"avg":3.7,"max":7.8,"min":0.4},"2":{"avg...   
1  {"1":{"avg":28,"max":30.8,"min":25.8},"2":{"av...   
2  {"1":{"avg":-2.5,"max":0.4,"min":-5.5},"2":{"a...   
3  {"1":{"avg":15.5,"max":22.8,"min":8.7},"2":{"a...   
4  {"1":{"avg":28.1,"max":32.5,"min":25.5},"2":{"...   

                       ideal_durations budget_level  culture  adventure  \
0            ["Short trip","One week"]       Luxury        5          2   
1             ["Long trip","One week"]       Luxury        2          4   
2  ["Short trip","Weekend","One week"]       Luxury        3          5   
3  ["Weekend","One week","Short trip"]    Mid-range        5          3   
4             ["Short trip","Weekend"]       Budget        4          3   

   nature  beaches  nightlife  cuisine  wellness  urban  seclusion  \
0       2        1          4        5         3      5          2   
1       5        5          2        3         4      1          5   
2       5        2          3        3         4      2          4   
3       3        1          3        4         3      4          2   
4       3        2          3        4         3      4          2   

   city_rating  annual_avg_temp_(C)  
0          3.2                   15  
1          3.4                   27  
2          3.4                    7  
3          3.1                   20  
4          3.1                   29

The original temperature column is a list of 12 monthly averages stored as a string.
This format is too detailed and awkward for analysis. Summarising the climate into a single annual average temperature allows us to:
- compare temperatures across cities
- analyse trends like “do warmer places tend to have higher budgets?”
- simplify the dataset for visualisations

**Ideal Durations Column**

We transformed the ideal_durations column to have proper lists using the literal_eval function.

In [92]:
df['ideal_durations'] = df['ideal_durations'].apply(ast.literal_eval)
df.head()

city    country         region  \
0           Milan      Italy         europe   
1  Yasawa Islands       Fiji        oceania   
2        Whistler     Canada  north_america   
3      Guanajuato     Mexico  north_america   
4        Surabaya  Indonesia           asia   

                                   short_description   latitude   longitude  \
0  Chic streets lined with fashion boutiques, his...  45.464194    9.189635   
1  Crystal-clear waters, secluded beaches, and vi... -17.290947  177.125786   
2  Snow-capped peaks and lush forests create a se...  50.117190 -122.954302   
3  Winding cobblestone streets and colorful facad...  20.987700 -101.000000   
4  Bustling streets filled with the aroma of loca...  -7.245972  112.737827   

                                    avg_temp_monthly  \
0  {"1":{"avg":3.7,"max":7.8,"min":0.4},"2":{"avg...   
1  {"1":{"avg":28,"max":30.8,"min":25.8},"2":{"av...   
2  {"1":{"avg":-2.5,"max":0.4,"min":-5.5},"2":{"a...   
3  {"1":{"avg":15.5,"max":22.8,"min":8.7},"2":{"a...   
4  {"1":{"avg":28.1,"max":32.5,"min":25.5},"2":{"...   

                   ideal_durations budget_level  culture  adventure  nature  \
0           [Short trip, One week]       Luxury        5          2       2   
1            [Long trip, One week]       Luxury        2          4       5   
2  [Short trip, Weekend, One week]       Luxury        3          5       5   
3  [Weekend, One week, Short trip]    Mid-range        5          3       3   
4            [Short trip, Weekend]       Budget        4          3       3   

   beaches  nightlife  cuisine  wellness  urban  seclusion  city_rating  \
0        1          4        5         3      5          2          3.2   
1        5          2        3         4      1          5          3.4   
2        2          3        3         4      2          4          3.4   
3        1          3        4         3      4          2          3.1   
4        2          3        4         3      4          2          3.1   

   annual_avg_temp_(C)  
0                   15  
1                   27  
2                    7  
3                   20  
4                   29

We then created new columns representing each trip duration. Then using Labada functions, we checked if the phrase was inside the list and if it was, the new column would be assigned a binary value of 1, if not, a binary value of 0.

In [93]:
df['is_short_trip'] = df['ideal_durations'].apply(lambda x: 1 if "Short trip" in x else 0)
df['is_one_week'] = df['ideal_durations'].apply(lambda x: 1 if "One week" in x else 0)
df['is_long_trip'] = df['ideal_durations'].apply(lambda x: 1 if "Long trip" in x else 0)
df['is_weekend'] = df['ideal_durations'].apply(lambda x: 1 if "Weekend" in x else 0)
df['is_day_trip'] = df['ideal_durations'].apply(lambda x: 1 if "Day trip" in x else 0)
df.head()

city    country         region  \
0           Milan      Italy         europe   
1  Yasawa Islands       Fiji        oceania   
2        Whistler     Canada  north_america   
3      Guanajuato     Mexico  north_america   
4        Surabaya  Indonesia           asia   

                                   short_description   latitude   longitude  \
0  Chic streets lined with fashion boutiques, his...  45.464194    9.189635   
1  Crystal-clear waters, secluded beaches, and vi... -17.290947  177.125786   
2  Snow-capped peaks and lush forests create a se...  50.117190 -122.954302   
3  Winding cobblestone streets and colorful facad...  20.987700 -101.000000   
4  Bustling streets filled with the aroma of loca...  -7.245972  112.737827   

                                    avg_temp_monthly  \
0  {"1":{"avg":3.7,"max":7.8,"min":0.4},"2":{"avg...   
1  {"1":{"avg":28,"max":30.8,"min":25.8},"2":{"av...   
2  {"1":{"avg":-2.5,"max":0.4,"min":-5.5},"2":{"a...   
3  {"1":{"avg":15.5,"max":22.8,"min":8.7},"2":{"a...   
4  {"1":{"avg":28.1,"max":32.5,"min":25.5},"2":{"...   

                   ideal_durations budget_level  culture  ...  wellness  \
0           [Short trip, One week]       Luxury        5  ...         3   
1            [Long trip, One week]       Luxury        2  ...         4   
2  [Short trip, Weekend, One week]       Luxury        3  ...         4   
3  [Weekend, One week, Short trip]    Mid-range        5  ...         3   
4            [Short trip, Weekend]       Budget        4  ...         3   

   urban  seclusion  city_rating  annual_avg_temp_(C)  is_short_trip  \
0      5          2          3.2                   15              1   
1      1          5          3.4                   27              0   
2      2          4          3.4                    7              1   
3      4          2          3.1                   20              1   
4      4          2          3.1                   29              1   

   is_one_week  is_long_trip  is_weekend  is_day_trip  
0            1             0           0            0  
1            1             1           0            0  
2            1             0           1            0  
3            1             0           1            0  
4            0             0           1            0  

[5 rows x 25 columns]

The original column stores multiple trip lengths in one cell (e.g., ["Short trip","Weekend","One week"]). This is not suitable for analysis because you can’t filter, group, or count these categories easily. With the mixed mixed list now separated into individual binary columns we can: 
- analyse which trip types are most common
- correlate trip duration with budget or other features
- build a cleaner dashboard (filters)

**Saving The Cleaned Dataset for Dashboarding**

Some elements that are needed for dashboarding will be different, or absent in the second dataset for analysis.

In [ ]:
# df_path = 'data/wtc_cleaned_dashboard_dataset.csv'

# # Saving the file using the df_path
# df.to_csv(df_path, index=False)

**Deleting Unnecessary Columns for Analysis Dataset**

We will be dropping the following columns:
- short_description
- avg_temp_monthly
- ideal_durations
- latitude
- longitude

They will be useful for when we create our Streamlit app. However, they are useless for analysis, so we will dropping deleting them.

In [95]:
# dropping the following columns as they're not needed for analysis
df.drop(columns=['short_description', 'avg_temp_monthly', 'ideal_durations', 'latitude', 'longitude'], inplace=True)
display(df.head())


city    country         region budget_level  culture  adventure  \
0           Milan      Italy         europe       Luxury        5          2   
1  Yasawa Islands       Fiji        oceania       Luxury        2          4   
2        Whistler     Canada  north_america       Luxury        3          5   
3      Guanajuato     Mexico  north_america    Mid-range        5          3   
4        Surabaya  Indonesia           asia       Budget        4          3   

   nature  beaches  nightlife  cuisine  wellness  urban  seclusion  \
0       2        1          4        5         3      5          2   
1       5        5          2        3         4      1          5   
2       5        2          3        3         4      2          4   
3       3        1          3        4         3      4          2   
4       3        2          3        4         3      4          2   

   city_rating  annual_avg_temp_(C)  is_short_trip  is_one_week  is_long_trip  \
0          3.2                   15              1            1             0   
1          3.4                   27              0            1             1   
2          3.4                    7              1            1             0   
3          3.1                   20              1            1             0   
4          3.1                   29              1            0             0   

   is_weekend  is_day_trip  
0           0            0  
1           0            0  
2           1            0  
3           1            0  
4           1            0

**Encoding Budget Level Column**

We're encoding the ```budget_level``` column using dictionary mapping to make it usable for analysis.

In [96]:
mapping = {"Budget": 1, "Mid-range": 2, "Luxury": 3}
df['budget_level'] = df['budget_level'].map(mapping)
display(df.head())

city    country         region  budget_level  culture  adventure  \
0           Milan      Italy         europe             3        5          2   
1  Yasawa Islands       Fiji        oceania             3        2          4   
2        Whistler     Canada  north_america             3        3          5   
3      Guanajuato     Mexico  north_america             2        5          3   
4        Surabaya  Indonesia           asia             1        4          3   

   nature  beaches  nightlife  cuisine  wellness  urban  seclusion  \
0       2        1          4        5         3      5          2   
1       5        5          2        3         4      1          5   
2       5        2          3        3         4      2          4   
3       3        1          3        4         3      4          2   
4       3        2          3        4         3      4          2   

   city_rating  annual_avg_temp_(C)  is_short_trip  is_one_week  is_long_trip  \
0          3.2                   15              1            1             0   
1          3.4                   27              0            1             1   
2          3.4                    7              1            1             0   
3          3.1                   20              1            1             0   
4          3.1                   29              1            0             0   

   is_weekend  is_day_trip  
0           0            0  
1           0            0  
2           1            0  
3           1            0  
4           1            0

Budget level was stored as categories (“Budget”, “Mid-range”, “Luxury”), which are strings. We converted these categories into numbers (e.g., 1, 2, 3) to: 
- perform correlations
- create numerical visuals (scatterplots, heatmaps)

---

## Section 4 — Research Question Analysis

### 4.1 — RQ1: Luxury Destinations & Temperature

In [97]:
display(df)

city         country         region  budget_level  culture  \
0             Milan           Italy         europe             3        5   
1    Yasawa Islands            Fiji        oceania             3        2   
2          Whistler          Canada  north_america             3        3   
3        Guanajuato          Mexico  north_america             2        5   
4          Surabaya       Indonesia           asia             1        4   
..              ...             ...            ...           ...      ...   
555            Maun        Botswana         africa             2        3   
556      Gothenburg          Sweden         europe             2        4   
557      Manchester  United Kingdom         europe             2        4   
558      Copenhagen         Denmark         europe             2        5   
559           Sucre         Bolivia  south_america             1        4   

     adventure  nature  beaches  nightlife  cuisine  wellness  urban  \
0            2       2        1          4        5         3      5   
1            4       5        5          2        3         4      1   
2            5       5        2          3        3         4      2   
3            3       3        1          3        4         3      4   
4            3       3        2          3        4         3      4   
..         ...     ...      ...        ...      ...       ...    ...   
555          5       5        1          2        3         3      2   
556          3       4        3          3        4         3      4   
557          2       2        1          4        4         3      4   
558          2       3        2          4        4         3      5   
559          3       3        1          2        3         2      2   

     seclusion  city_rating  annual_avg_temp_(C)  is_short_trip  is_one_week  \
0            2          3.2                   15              1            1   
1            5          3.4                   27              0            1   
2            4          3.4                    7              1            1   
3            2          3.1                   20              1            1   
4            2          3.1                   29              1            0   
..         ...          ...                  ...            ...          ...   
555          4          3.1                   24              1            1   
556          3          3.4                    9              1            1   
557          2          2.9                   10              1            1   
558          2          3.3                   10              1            1   
559          4          2.7                   16              1            1   

     is_long_trip  is_weekend  is_day_trip  
0               0           0            0  
1               1           0            0  
2               0           1            0  
3               0           1            0  
4               0           1            0  
..            ...         ...          ...  
555             0           0            0  
556             0           1            0  
557             0           1            0  
558             0           1            0  
559             0           1            0  

[560 rows x 20 columns]

### 4.2 — RQ2: Budget Level & Trip Duration

## Section 5 — General EDA

Broader, non-RQ-specific exploration:

- Global city distribution

- Region/country breakdown (bar chart or treemap)

- Rating comparisons (culture, nature, beaches, cuisine…)

- Scatter plots for relationships

- Summary insights

## Conclusions and Next Steps

Summarise what was accomplished in this notebook.
- State what needs to be done next, for example:

- Integrate outputs into Power BI/Tableau dashboard

- Start building Streamlit interface

- Create presentation slides

- Conduct team code review